src/middlewared/middlewared/plugins/container/container.py

```diff
@@ -258,3 +260,20 @@ def do_delete(self, id_):
         self.middleware.call_sync('pool.dataset.delete', container['dataset'])
 
         self.middleware.call_sync('etc.generate', 'libvirt_guests')
+
+    @api_method(ContainerPoolChoicesArgs, ContainerPoolChoicesResult, roles=['VIRT_GLOBAL_READ'])
+    async def pool_choices(self):
+        """
+        Pool choices for virtualization purposes.
+        """
+        pools = {}
+        imported_pools = await self.middleware.run_in_thread(query_imported_fast_impl)
+        for p in imported_pools.values():
+            if p['name'] in BOOT_POOL_NAME_VALID:
+                continue
+
+            ds = await self.middleware.call('pool.dataset.get_instance_quick', p['name'], {'encryption': True})
```

Comment:
Use zfs.resource.query_impl instead.

Reply:
How do I make it work with encryption?

Reply:
A simple grep of our code base would show you. You just add the "encryption" property similar to what you're doing now with pool.dataset. I think the sysdataset plugin has examples